pandas/core/tools/timedeltas.py

```diff
@@ -129,31 +126,27 @@ def _convert_listlike(arg, unit='ns', box=True, errors='raise', name=None):
     """Convert a list of objects to a timedelta index object."""
 
     if isinstance(arg, (list, tuple)) or not hasattr(arg, 'dtype'):
+        # This is needed only to ensure that in the case where we end up
+        #  returning arg (errors == "ignore"), and where the input is a
+        #  generator, we return a useful list-like instead of a
+        #  used-up generator
         arg = np.array(list(arg), dtype='O')
```

Comment:
-> object. We are not consistent across the codebase with specifiying object dtype (not a big deal as its a small thing), but small things add up

Reply:
done